scalanet/src/io/iohk/scalanet/peergroup/kademlia/KRouter.scala

```diff
@@ -14,19 +19,23 @@ import io.iohk.scalanet.peergroup.kademlia.KRouter.{Config, NodeRecord}
 import monix.eval.Task
 import monix.reactive.{Consumer, Observable, OverflowStrategy}
 import org.slf4j.LoggerFactory
+import org.spongycastle.crypto.params.ECPrivateKeyParameters
 import scodec.bits.BitVector
+import io.iohk.scalanet.crypto
 
 class KRouter[A](
     val config: Config[A],
     val network: KNetwork[A],
     private val routerState: Ref[Task, NodeRecordIndex[A]],
+    private val privateKey:ECPrivateKeyParameters,
     val clock: Clock = Clock.systemUTC(),
     val uuidSource: () => UUID = () => UUID.randomUUID(),
     val rnd: Random = new SecureRandom()
-) {
+)(implicit codec:Codec[A]) {
 
   private val log = LoggerFactory.getLogger(getClass)
 
+  var actualNodeRecord = config.nodeRecord
```

Comment:
it is safe to use var here? 

(rename: `actualNodeRecord` to `currentNodeRecord`)